bson/src/main/org/bson/types/Decimal128.java

```diff
@@ -374,6 +383,57 @@ public boolean isNaN() {
     }
 
 
+    @Override
+    public int compareTo(final Decimal128 o) {
```

Comment:
If anyone can see a way to simplify this, I'm all for it.

Reply:
Does this need a null check?

Reply:
It will throw NPE on first access of o, which is what the contract of compareTo requires